torch/nn/utils/rnn.py

```diff
@@ -123,3 +124,106 @@ def pad_packed_sequence(sequence, batch_first=False, padding_value=0.0):
     if batch_first:
         output = output.transpose(0, 1)
     return output, lengths
+
+
+def pad_sequence(sequences, lengths, batch_first=False):
+    r"""Pad a list of variable length Variables with zero
+
+    The ``pad_sequence`` pads the list of Variables on zeroth dimension and
+    stack all the sequences on zeroth dimension. For example, if the input is
+    list of sequences with size `` Lx*`` and if batch_first is False, the
+    output will be of size `` TxBx* `` and if batch_first is True,
+    output will be of size ``BxTx* ``. The ``pad_sequence`` accepts list of
+    sequences and its lengths which should be sorted in decreasing order
+
+    B is batch size
+    T is length longest sequence
+    L is length of the sequence
+    * is any trailing dimension including zero
+
+    >>> from torch.nn.utils.rnn import pad_sequence
+    >>> a = Variable(torch.ones(25, 300))
+    >>> b = Variable(torch.ones(15, 300))
+    >>> c = Variable(torch.ones(22, 300))
+    >>> pad_sequence([a, b, c], [25, 15, 22]).size()
+    torch.Size([25, 3, 300])
+
+    Note:
+        This function returns a Variable of size LxBx* or BxLx* where L is the
+        length of longest sequence (lengths[0])
+
+    Arguments:
+        sequences (list(Variable)): list of variable length sequences.
+        lengths (list[int]): list of sequences lengths of each batch element.
+        batch_first (bool, optional): if True, the input is expected in Bx*x*
+            format.
+
+    Returns:
+        Variable of size ``seq_len x len(sequences) x * ``
+            if batch_first = False
+        Variable of size ``len(sequences) x seq_len x * `` otherwise
+    """
+
+    if len(lengths) != len(sequences):
+        raise ValueError("number of elements in lengths and sequences didn't match")
+
+    out_variable = []
+    max_len = lengths[0]
+    prev_l = lengths[0]
+    for variable, length in zip(sequences, lengths):
+        # temperory sort check, will be removed when we handle sorting internally
+        if prev_l < length:
+                raise ValueError("lengths array has to be sorted in decreasing order")
+        prev_l = length
+        if length < max_len:
+            prev_l = length
+            padding_shape = [lengths[0] - length] + list(variable.size()[1:])
+            filler = Variable(torch.Tensor(*padding_shape).type_as(variable.data).zero_())
+            out_variable.append(torch.cat((variable, filler)))
+        else:
+            out_variable.append(variable)
+    if batch_first:
+        return torch.stack(out_variable)
+    else:
+        return torch.stack(out_variable).transpose(0, 1)
```

Comment:
this if can be replaced with `return torch.stack(out_variable, 0 if batch_first else 1)`